bcda/auth/client/oktaclient.go

```diff
@@ -149,6 +163,54 @@ func (oc *OktaClient) AddClientApplication(localID string) (string, string, erro
 	return clientID, clientSecret, nil
 }
 
+func (oc *OktaClient) RequestAccessToken(creds Credentials) (OktaToken, error) {
+	requestID := uuid.NewRandom()
+
+	tokenURL := fmt.Sprintf("%s/oauth2/%s/v1/token", oktaBaseUrl, oktaServerID)
+	req, err := http.NewRequest("POST", tokenURL, nil)
+	if err != nil {
+		return OktaToken{}, err
+	}
+
+	req.Header.Add("Accept", "application/json")
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Add("Cache-Control", "no-cache")
+
+	params := url.Values{}
+	params.Set("client_id", creds.ClientID)
+	params.Set("client_secret", creds.ClientSecret)
+	params.Set("grant_type", "client_credentials")
+	params.Set("scope", "bcda_api")
+	req.URL.RawQuery = params.Encode()
+
+	httpClient := &http.Client{}
+
+	logRequest(requestID).Print("requesting Okta access token")
+	resp, err := httpClient.Do(req)
```

Comment:
one more place you can use `client()`.  that'll also ensure it has a timeout

Reply:
+1